### PR TITLE
Pythia test

### PR DESCRIPTION
**Pull request by joanvelja**
Ok for PR: 

`pythia.py` is the main file that takes in as parser args:

```
parser.add_argument("--lr", type=float, default=5e-5)
    parser.add_argument("--num_epochs", type=int, default=2)
    parser.add_argument(
        "--model_name", type=str, default="EleutherAI/pythia-70m"
    )  # Switch here for different models
    parser.add_argument(
        "--enable_BSGC", type=bool, default=False
    )  # defaulting to False for our purposes
```

supports all pythia models, saves locally ckpoints at end of epochs and pushes to hub latest trained checkpoint  + tokenizer. 

Can find pythia70m checkpoint resulting from this training run here: https://huggingface.co/jvelja/pythia-finetune-pythia-70m-clusters-4-NoBSGC/tree/main

And 1.4b coming soon.

### CODE REVIEW

**Comment by joanvelja** [comment]
## Updated way to run after new commit

### Basic Usage
```
python pythia.py --model_name EleutherAI/pythia-70m --lr 5e-5 --num_epochs 2
```

**Optional Arguments**

- --model_name: The Hugging Face model name (default: "EleutherAI/pythia-70m")
- --lr: Learning rate (default: 5e-5)
- --num_epochs: Number of training epochs (default: 2)
- --enable_BSGC: Enable BSGC (requires --do_modularity)
- --do_modularity: Enable modularity analysis
- --mix_data: Train on both RAVEL and WIKI datasets


### Running Pythia-1.4B Experiments
Note: You need to have set up your Hugging Face token in the environment variables before running the script.

**Training Commands**
1. **Pythia-1.4B with Modularity + Mixed Data (RAVEL + Wiki)**
```
python pythia.py \
    --model_name EleutherAI/pythia-1.4b \
    --lr 1e-4 \
    --num_epochs 2 \
    --do_modularity \
    --mix_data
```

2. **Pythia-1.4B with Modularity and on Wiki Data only**
```
python pythia.py \
    --model_name EleutherAI/pythia-1.4b \
    --lr 1e-4 \
    --num_epochs 2 \
    --do_modularity
```

3. **Pythia-1.4B with Mixed Data (No Modularity)**
```
python pythia.py \
    --model_name EleutherAI/pythia-1.4b \
    --lr 1e-4 \
    --num_epochs 2 \
    --mix_data
```

4. **Pythia-1.4B Base Training (No Modularity, Wiki only)**
```
python pythia.py \
    --model_name EleutherAI/pythia-1.4b \
    --lr 1e-4 \
    --num_epochs 2
```

**Outputs**
- Models will be pushed to HuggingFace
- SVD dictionaries in local
- Metrics saved in local
